example/mixed-arch/main.tf

```diff
@@ -2,22 +2,22 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "=2.51.0"
+      version = ">=2.51.0"
```

Comment:
Can we replace the `>=` operator here with `~>` to illustrate best practices in the examples? At the same time, you can drop the last bit of the version e.g. `~> 2.51`, which tells Terraform that any version >= 2.51 but <3.0 is fine.

In reusable modules, we want to use the `>=` operator to set the minimum required by the API, but in root modules, we want to use `~>` to avoid version upgrades that may introduce breaking changes.

Also, shouldn't this require v2.57.0+ due to the UAI fix?